tokio/tests/no_rt.rs

```diff
@@ -0,0 +1,26 @@
+use tokio::time::{timeout, Duration};
+
+#[test]
+#[should_panic(expected = "no timer running")]
+fn panics_when_no_timer() {
+    use futures::executor::block_on;
+    block_on(timeout_value());
+}
+
+async fn timeout_value() {
+    use tokio::sync::oneshot;
+
+    let (_tx, rx) = oneshot::channel::<()>();
+
+    let dur = Duration::from_millis(20);
+
+    let _ = timeout(dur, rx).await;
+}
+
+// #[test]
```

Comment:
I assume we can remove this commented out test?

Reply:
we can - i just included it for now to show my thought process for such a test. like i said above - not quite sure how to test the other panic types, only one i could get was a timer panic

Reply:
Ah yup, sorry I totally missed that and went to the code right away! 

So I think it should be possible to catch the io driver panic if you try to tcp connect? If not I am fine not testing all of them for now. If that is the case id like to remove the commented out test then we can merge this. Thanks!